common/memstream.h

```diff
@@ -342,6 +342,96 @@ class MemoryReadWriteStream : public SeekableReadStream, public SeekableWriteStr
 	byte *getData() { return _data; }
 };
 
+/**
+ * A seekable read and writeable memory stream that operates on an already existing memory buffer
+ */
+class MemorySeekableReadWriteStream : public SeekableReadStream, public SeekableWriteStream {
+private:
+	const uint32 _bufSize;
+	byte *_ptrOrig;
+	byte *_ptr;
+	uint32 _pos;
+	bool _err;
+	bool _eos;
+
+public:
+	MemorySeekableReadWriteStream(byte *buf, uint32 len) : _ptrOrig(buf), _ptr(buf), _bufSize(len), _pos(0), _err(false), _eos(false) {}
```

Comment:
Maybe for consistency you should have a `DisposeAfterUse::Flag` here as well as we do for other steam classes (and if it is `YES` free the `_ptrOrig` in the destructor)?

Reply:
Added the flag. Thanks